### PR TITLE
xdg-mime xdg-open xdg-settings xprop

### PR DESCRIPTION
**Pull request by beroal**
When I execute `xdg-settings set default-url-scheme-handler https mpv.desktop`, I get the following log messages:

```
apparmor="ALLOWED" operation="file_mmap" class="file" profile="xdg-settings//bus" name="/usr/bin/dbus-send"  comm="dbus-send" requested_mask="r" denied_mask="r" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="file_inherit" class="file" profile="xprop" name="/dev/pts/2"  comm="xprop" requested_mask="wr" denied_mask="wr" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
apparmor="DENIED" operation="file_inherit" class="file" profile="xdg-mime" name="/dev/pts/2"  comm="xdg-mime" requested_mask="wr" denied_mask="wr" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
apparmor="ALLOWED" operation="file_mmap" class="file" profile="xdg-mime//bus" name="/usr/bin/dbus-send"  comm="dbus-send" requested_mask="r" denied_mask="r" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="open" class="file" profile="xdg-mime" name="/proc/version"  comm="grep" requested_mask="r" denied_mask="r" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="mkdir" class="file" profile="xdg-settings" name="/home/user/.local/share/applications/"  comm="mkdir" requested_mask="c" denied_mask="c" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
apparmor="DENIED" operation="mknod" class="file" profile="xdg-mime" name="/home/user/.config/mimeapps.list"  comm="touch" requested_mask="c" denied_mask="c" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
apparmor="DENIED" operation="mknod" class="file" profile="xdg-mime" name="/home/user/.config/mimeapps.list.new"  comm="xdg-mime" requested_mask="c" denied_mask="c" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
```

After adding the corresponding AppArmor rules, I get the following log messages:

```
apparmor="DENIED" operation="rename_src" class="file" profile="xdg-mime" name="/home/user/.config/mimeapps.list.new"  comm="mv" requested_mask="r" denied_mask="r" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
```

After adding the corresponding AppArmor rules, no log messages.

Also, when I execute `xdg-settings set default-url-scheme-handler https firefox-chat.desktop` (no `firefox-chat.desktop` in `/usr/share/applications`), I get the following log messages:

```
apparmor="DENIED" operation="mknod" class="file" profile="xdg-settings" name="/home/user/.local/share/applications/firefox-chat.desktop.Oi9XB5"  comm="mktemp" requested_mask="c" denied_mask="c" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
```

After adding the corresponding AppArmor rules, no log messages.

When I execute `xdg-open $URL`, I get the following log messages:

```
apparmor="DENIED" operation="file_inherit" class="file" profile="xprop" name="/dev/pts/2"  comm="xprop" requested_mask="wr" denied_mask="wr" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
apparmor="DENIED" operation="file_inherit" class="file" profile="xdg-mime" name="/dev/pts/2"  comm="xdg-mime" requested_mask="wr" denied_mask="wr" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
apparmor="DENIED" operation="exec" class="file" profile="xdg-settings" name="/usr/bin/sleep"  comm="xdg-settings" requested_mask="x" denied_mask="x" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="open" class="file" profile="xdg-settings" name="/usr/bin/sleep"  comm="xdg-settings" requested_mask="r" denied_mask="r" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="file_inherit" class="file" profile="xdg-open//bus" name="/dev/pts/2"  comm="dbus-send" requested_mask="wr" denied_mask="wr" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
apparmor="DENIED" operation="open" class="file" info="Failed name lookup - disconnected path" error=-13 profile="xdg-open//bus" name="apparmor/.null"  comm="dbus-send" requested_mask="wr" denied_mask="wr" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="file_mmap" class="file" profile="xdg-open//bus" name="/usr/bin/dbus-send"  comm="dbus-send" requested_mask="r" denied_mask="r" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="open" class="file" profile="xdg-open" name="/proc/version"  comm="grep" requested_mask="r" denied_mask="r" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="exec" class="file" profile="xdg-open" name="/usr/bin/env"  comm="xdg-open" requested_mask="x" denied_mask="x" fsuid=1001 ouid=0 FSUID="user" OUID="root"
apparmor="DENIED" operation="open" class="file" profile="xdg-open" name="/usr/bin/env"  comm="xdg-open" requested_mask="r" denied_mask="r" fsuid=1001 ouid=0 FSUID="user" OUID="root"
```

After adding the corresponding AppArmor rules, I get the following log messages:

```
apparmor="DENIED" operation="exec" class="file" profile="xdg-open" name="/usr/lib/firefox/firefox"  comm="env" requested_mask="x" denied_mask="x" fsuid=1001 ouid=0 FSUID="user" OUID="root"
```

**I guess that all GUI applications that aren't `DBusActivatable` should be added to the `xdg-open` profile. I don't know how to do this.**

### CODE REVIEW

File: apparmor.d/groups/freedesktop/xdg-open [hunk 1]
```diff
@@ -35,10 +36,15 @@ profile xdg-open @{exec_path} flags=(attach_disconnected) {
...
-  profile bus {
+  @{PROC}/version r,
+
+  profile bus flags=(attach_disconnected) {
     include <abstractions/base>
     include <abstractions/app/bus>
     include <abstractions/bus-session>
+    include <abstractions/consoles>
+
+    @{bin}/dbus-send mr,
```
Comment (roddhjav) [inline apparmor.d/groups/freedesktop/xdg-open:47]: `@{bin}/dbus-send mr,` is already part of `abstractions/app/bus`

**Comment by roddhjav** [comment]
Thanks, I have recently fully rewritten these profile. However, it seems they do need much more test on various DE.

In general, the way we open resources should be revisited and **much more** restricted.

**Comment by roddhjav** [comment]
> I guess that all GUI applications that aren't DBusActivatable should be added to the xdg-open profile. 

As this profile is only used interactively (never as a dependencies of other program), this may be the solution. It also really mean that this profile should not be used by any apparmor profiles.

> I don't know how to do this.

Simply add: `include <abstractions/app-open>` and it will become a generic equivalent of `child-open`

**Comment by beroal** [comment]
> As this profile is only used interactively (never as a dependencies of other profile), this may be the solution. It also really mean that this profile should not be used by any apparmor profiles.

I'm not sure why you say that the `xdg-open` profile should not be used by any apparmor profile. `xdg-open` is in the `open_names` variable which is in the `open_path` variable which is [recommended for opening resources](https://apparmor.pujol.io/development/internal/#open-resources).

**Comment by roddhjav** [comment]
`@{open_path}` is intended to be used as follow:
```
@{open_path} rPx -> child-open*,
```

Never like this: `@{open_path} rPx,`

Like this, you can control the kind of resource a program can open.

**Comment by roddhjav** [comment]
Thanks, merged.

**Pull request merged**